nemo_text_processing/text_normalization/normalize_with_audio.py

```diff
@@ -94,14 +95,10 @@ def normalize_with_audio(self, text: str, verbose: bool = False) -> str:
             if verbose:
                 print(text)
             return text
-        text = pynini.escape(text)
-
-        def get_tagged_texts(text):
-            tagged_lattice = self.find_tags(text)
-            tagged_texts = self.select_all_semiotic_tags(tagged_lattice)
-            return tagged_texts
 
-        tagged_texts = set(get_tagged_texts(text))
+        text = pynini.escape(text)
+        # TODO add preprocess?
+        tagged_texts = rewrite.top_rewrites(text, self.tagger.fst, nshortest=100)
```

Comment:
using `nshortest=100` here yields poor results with a sentence like:
`“If 15,000 men can be spared for the flank movement, certainly 10,000 may be sent to reinforce General Lee.”`

The 2nd "10,000" will never be verbalized into "ten thousand" because it never appears in tagged_texts. A number like `nshortest=1000` seems to work at the expense of processing time.

It's a relatively straight-forward sentence to fail on. Maybe normalize() could accept nshortest as a parameter since cleaner results may be required?